### PR TITLE
MWPW-134131 - Fixing issue when there are more than one rule with same number of grouping

### PR DESCRIPTION
**Pull request by sabyamon**
Fixing the issue of the first rule getting picked when there was more than one rule with the same number of grouping.

Resolves: https://jira.corp.adobe.com/browse/MWPW-134131

**Test URLs:**
- Before: https://uar-integration--milo--adobecom.hlx.page/drafts/sabya/quiz-issue/?martech=off&hideGeorouting=on
- After: https://uar-same-number-group--milo--sabyamon.hlx.page/drafts/sabya/quiz-issue/?martech=off&hideGeorouting=on

### CODE REVIEW

File: libs/blocks/quiz/utils.js [hunk 1]
```diff
@@ -317,24 +317,28 @@ export const findMatchForSelections = (results, selections) => {
...
   // Case 2 - when you have clauses grouped with parenthesis.
-  const compoundResults = results.find((destination) => {
-    if (destination.result.indexOf('(') !== -1 && destination.result.split('&').length === userSelectionLen) {
-      return destination;
+  const probableMatches = [];
+  results.forEach((rule) => {
+    if (rule.result.indexOf('(') !== -1 && rule.result.split('&').length === userSelectionLen) {
+      probableMatches.push(rule);
     }
   });
```
Comment (chrischrischris) [inline libs/blocks/quiz/utils.js:325]: nit
```suggestion
const probableMatches = results.reduce((match, rule) => {
  if (rule.result.includes('(') && rule.result.split('&').length === userSelectionLen) {
    match.push(rule);
  }
  return match;
}, []);
```

File: libs/blocks/quiz/utils.js [hunk 2]
```diff
@@ -317,24 +317,28 @@ export const findMatchForSelections = (results, selections) => {
...
-
-  if (productList.length) {
-    const isCompoundProductsMatched = selections.primary.every(
-      (product, index) => productList[index].includes(product),
-    );
-    if (isCompoundProductsMatched) {
-      recommendations.push(compoundResults);
-      return recommendations;
+  probableMatches.forEach((rule) => {
+    const productList = rule !== undefined ? rule.result.split('&') : [];
```
Comment (chrischrischris) [inline libs/blocks/quiz/utils.js:328]: nit: if `rule` is never `''` then you can just do `rule ? rule.result.split...`

**Review by chrischrischris — APPROVED** [review]

**Review by mokimo — APPROVED** [review]
Would be nice if you can fix some of the linting errors in this file (or skip the linter for the specific line if not otherwise possible)

**Review by overmyheadandbody — APPROVED** [review]
`console.log` call from L52 should probably be switched to a LANA log, same as any other calls to `console`

**Review by narcis-radu — APPROVED** [review]
Chris and Rareș have already left some suggestions, it would be great to implement them before merging. Thank you!

**Review by elaineskpt — APPROVED** [review]

**Review by JackySun9 — APPROVED** [review]

File: libs/blocks/quiz/utils.js (hunk 1 above)
Comment (sabyamon) [reply libs/blocks/quiz/utils.js]: Looks much better. Thanks for the feedback. Updated.

File: libs/blocks/quiz/utils.js (hunk 2 above)
Comment (sabyamon) [reply libs/blocks/quiz/utils.js]: Thanks Chris! updated.

**Comment by sabyamon** [comment]
> Would be nice if you can fix some of the linting errors in this file (or skip the linter for the specific line if not otherwise possible)

Thanks @mokimo for the review! Updated and made the linter happy for this file.

**Comment by sabyamon** [comment]
> Chris and Rareș have already left some suggestions, it would be great to implement them before merging. Thank you!

Addressed all the review comments. It's immensely helping us before going to the `main` branch. Thanks all!

**Pull request merged**